Firmware/Marlin_main.cpp

```diff
@@ -5241,16 +5241,42 @@ void process_commands()
     }
 
     /*!
-    ### M78 - Show statistical information about the print jobs <a href="https://reprap.org/wiki/G-code#M78:_Show_statistical_information_about_the_print_jobs">M78: Show statistical information about the print jobs</a>
+    ### M78 - Get/set statistics <a href="https://reprap.org/wiki/G-code#M78:_Show_statistical_information_about_the_print_jobs">M78: Show statistical information about the print jobs</a>
+
+    #### Usage
+
+        M78 [ S | T ]
+
+    #### Parameters
+    - `S` - Set used filament length in cm
+    - `T` - Set total print time in minutes
     */
     case 78:
     {
         // @todo useful for maintenance notifications
-        SERIAL_ECHOPGM("STATS ");
-        SERIAL_ECHO(eeprom_read_dword((uint32_t *)EEPROM_TOTALTIME));
-        SERIAL_ECHOPGM(" min ");
-        SERIAL_ECHO(eeprom_read_dword((uint32_t *)EEPROM_FILAMENTUSED));
-        SERIAL_ECHOLNPGM(" cm.");
+        const char *_m_fil;
+        const char *_m_time;
+        uint32_t _cm = 0;
+        uint32_t _min = 0;
+
+        if (printJobOngoing()) {
+          _m_fil = _O(MSG_FILAMENT_USED);
+          _m_time = _O(MSG_PRINT_TIME);
+          _cm = ((uint32_t)total_filament_used) / (1000);
+          _min = (print_job_timer.duration() / 60);
```

Comment:
Nitpick: Extra bracket.
You can safely change `(print_job_timer.duration() / 60);` to `print_job_timer.duration() / 60;`